### PR TITLE
Add ``unrecognized-option`` message

### LINKED ISSUE #5259: Proper warning when the ``pyproject.toml`` options do not match existing options
### Current problem

We're not going to crash following the merging of #4720 but we need to define better what is expected in the pyproject.toml for pylint. When using:

```toml
[tool.pylint]
load-plugins = []
disable = "logging-not-lazy,logging-format-interpolation"
```
``load-plugins`` or ``disable`` are not valid options in the main namespace.

The expected configuration would be:

```toml
[tool.pylint]
    [tool.pylint."messages control"]
        disable = "logging-not-lazy,logging-format-interpolation"
    [tool.pylint."master"]
        load-plugins = []
```

But we're not warning the user about it.

### Desired solution

 We need to have proper error message when something is misplaced, because right now it's not very convenient. We're not going to crash following the merging of #4720 but we also should have an error message like:

``disable is not a valid section for pylint, did you mean '[tool.pylint."messages control"]  disable = ...' ?``

### Additional context

https://github.com/PyCQA/pylint/pull/4720#issuecomment-886204285

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.
- [x] If you used multiple emails or multiple names when contributing, add your mails
   and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :sparkles: New feature |

## Description

Closes #5259.

The behaviour does not really change but is a little more elegant.
1) For unrecognised options in a configuration file we know warn about that error and display a message but continue parsing.
2) For unrecognised options in the command line we know exit with a related exception. Previously we would crash with "don't recognise module --unknown-option" so in terms of behaviour not much changes, but it is just a little more elegant.

I also removed all unused options in our own configuration files as it failed our tests with this new message.

### CODE REVIEW

**Comment by DanielNoord** [comment]
Going to add an additional test. Drafting.

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas Ready for review again!

File: pylintrc [hunk 1]
```diff
@@ -324,36 +303,26 @@ inlinevar-naming-style=any
...
 method-naming-style=snake_case
 
 # Regular expression matching correct method names
 method-rgx=[a-z_][a-z0-9_]{2,}$
 
-# Naming hint for method names
-method-name-hint=[a-z_][a-z0-9_]{2,}$
-
 # Regular expression which can overwrite the naming style set by typevar-naming-style.
 #typevar-rgx=
```
Comment (Pierre-Sassoulas) [inline pylintrc:328]: The cleanup of our own pylintrc is pretty great !

**Review by Pierre-Sassoulas — APPROVED** [review]
🎉

**Pull request merged**